src/ivalue_manager.rs

```diff
@@ -268,8 +284,19 @@ impl<'a> IValueKeyHolderWrite<'a> {
     }
 
     fn get_json_holder(&mut self) -> Result<(), RedisError> {
-        if self.val.is_none() {
-            self.val = self.key.get_value::<RedisJSON<IValue>>(&REDIS_JSON_TYPE)?;
+        if self.value.is_none() {
+            self.value = match self.key.key_type() {
+                KeyType::Module => self
+                    .key
+                    .get_value::<RedisJSON<IValue>>(&REDIS_JSON_TYPE)?
```

Comment:
Could have returned an Error and now will return `Ok`?

Reply:
it will still return an error on anything that is not a JSON or String that be parsed as a JSON